access-ocp-api-through-cis.md

```diff
@@ -0,0 +1,104 @@
+## Accessing the api endpoint deployed on OpenShift through CIS
+
+1. Setup an OpenShift cluster with ingress. The cluster comes with a default endpoint `/healthz` that is accessible as:
+   ```
+   % oc get routes -n openshift-ingress
+   NAME                                      HOST/PORT                                                                               PATH       SERVICES                  PORT   TERMINATION     WILDCARD
+   route.route.openshift.io/router-default   router-default.xxx-3b5bf5f75xxxx21c8c35ad277-0000.us-south.containers.appdomain.cloud   /healthz   router-internal-default   1936   edge/Redirect   None
+
+   % curl https://router-default.xxx-3b5bf5f75xxxx21c8c35ad277-0000.us-south.containers.appdomain.cloud/healthz
+   ok
+   ```
+
+   Alternatively you deploy your app and create the NodePort service for the application endpoint.
+
+2. Open the CIS service and navigate to `DNS` tab under `Reliability`.
+
+3. Go to `DNS records` and add a new record:
+   ```
+   Type: CNAME
+   Name: <any name>
+   TTL: Automatic
+   Alias domain name: <openshift_route>  ## example: router-default.xxx-3b5bf5f75xxxx21c8c35ad277-0000.us-south.containers.appdomain.cloud
+   ```
+
+4. If CIS domain is `example.com` and DNS record name provided in step 3 is `test` then the endpoint can be accessed via `https://test.example.com/<api_endpoint>`. If you access the link now, you will get `ssl handshake failure` error as below.
+   ```
+   % curl -v https://test.example.com/<api_endpoint>
+   *   Trying [2606:4000:10::xy43:xy0]:443...
+   * Connected to test.example.com (2606:4000:10::xy43:xy0) port 443 (#0)
+   * ALPN: offers h2,http/1.1
+   * (304) (OUT), TLS handshake, Client hello (1):
+   *  CAfile: /etc/ssl/cert.pem
+   *  CApath: none
+   * LibreSSL/3.3.6: error:1404B410:SSL routines:ST_CONNECT:sslv3 alert handshake failure
+   * Closing connection 0
+   curl: (35) LibreSSL/3.3.6: error:1404B410:SSL routines:ST_CONNECT:sslv3 alert handshake failure
+   ```
+
+   It is because the client or server is not able to establish a secure connection.
+
+
+5. To establish a secure connection between client and server, you need to have appropriate SSL certificates. The SSL certificates can be generated using [Secrets Manager](https://cloud.ibm.com/catalog/services/secrets-manager) service on IBM Cloud. Order a certificate in Secrets Manager:
```

Comment:
Seems like a prereq? If so, put it under a `## Before you begin` section before the steps. seems like steps 5 and 6 are prereqs. Step 7 would refer to the cert you downloaded.

Reply:
There should be two pre-req for this - 1. CIS instance with a valid domain being available 2. Openshift deployed with an application and endpoint available to access. In this you can mention that this doc uses an existing /healthz endpoint as an example.

CIS CNAME is not a pre-req.  But how we should word it is, CNAME is created and the application will not be still accessible as the certificates are not set for https protocol.  And then provide the steps for the certificate setup.  (Something similar to these wordings need not be the exact ones)

Reply:
I dont think we should paste the entire ssl handshake error in this case but just mention that the SSL certificates should be set before the endpoint is accessed.

Reply:
Updated doc.